build-gluster-org/scripts/user-update.py

```diff
@@ -0,0 +1,114 @@
+                continue
+            else:
+                NOT_FOUND.append("{}  ".format(INDENTATION_VALUE)+"- "+username['login'])
+    f.close()
+    return NOT_FOUND
+
+#Function which appends the names which are not present in the jenkins conf file
+def edit_file(repo_path,filename,pattern,NOT_FOUND):
+    os.chdir(repo_path)
+    with open(filename, "r") as input_file:
+        buffer = input_file.readlines()
+
+    with open(filename, "w") as output_file:
+        search_pattern = re.compile(r'\s*%s'%pattern) 
+        for line in buffer:
+            if search_pattern.search(line):  
+                for names in NOT_FOUND:
+                    line = line + str(names) + "\n"
+            output_file.write(line)
+
+    input_file.close()
+    output_file.close()
+
+#Checks the files which are modified
+def check_untracked_files(repo_path,pat,branch_name):
+    os.chdir(repo_path)
+    FILES = []
+    os.system('git status')
+    out = subprocess.Popen(['git', 'status'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    stdout, stderr = out.communicate()
+    output_decoded = stdout.decode('UTF-8').split('\n')
+    for line in output_decoded:
+         if pat in line:
+             val = line.split(':')
+             FILES.append(val[1].strip())
+    print (FILES)
+    if len(FILES) == 0:
+        print ("No changes")
+    else:
+        commit_and_push(FILES,branch_name)
+
+#Commit the files and push it to the repo
+def commit_and_push(FILENAMES,branch_name):
+    file_names=' '.join(FILENAMES)
+    print ("Changed files : {}".format(file_names))
+    os.system('git add {}'.format(file_names))
+    os.system('git commit -m "Updating the admin-list as per gluster-maintainers list"')
+    os.system('git push origin {}'.format(branch_name))
+
+#Creates pull request against the master branch
+def create_pull_request(username, password, repo_name, title, description, head_branch, base_branch):
+    git_pulls_api = "https://api.github.com/repos/gluster/{}/pulls".format(repo_name)
+    payload = { "title": title, "body": description, "head": head_branch, "base": base_branch }
+    r = requests.post( git_pulls_api, auth=(username,password), data=json.dumps(payload))
+    if r.ok:
+        print("Pull request successfully created!")
+        pull_requests = requests.get("https://api.github.com/repos/gluster/{}/pulls".format(repo_name), auth=(username, password)).json()
+        for pull in pull_requests:
+             if pull['title'] == title:
+                 print (pull['html_url'])
+    else:
+        print("Pull request Failed: {}".format(r.text))
+
+def main():
+    username = os.environ['GITHUB_USERNAME']
+    password = os.environ['GITHUB_PASSWORD']
+    url = f"https://api.github.com/orgs/gluster/teams/gluster-maintainers/members"
+    files = os.environ['JOB_DETAILS'].rstrip()
+    pattern = "admin-list:"
+    pattern2 = "modified:"
+    dev_branch_name = "user-update-branch"
```

Comment:
Should we have different branch name, since this will be potentially used for multiple pull requests ?

Like, we modify the list of user on friday, the job do a PR, we do not merge it because that's friday and we are too busy partying alone, or any relaxing activity we can do, and then someone modify again one day after (we still do not work). 

Does it create another PR, or replace the existing ? 

Also, what happen if the PR is merged ? (that's a real question, I never did that as I usually do a new branch for each PR, so I genuinely do not know :/ )